src/cdk/platform/features/scrolling.ts

```diff
@@ -28,9 +28,38 @@ export const enum RtlScrollAxisType {
 /** Cached result of the way the browser handles the horizontal scroll axis in RTL mode. */
 let rtlScrollAxisType: RtlScrollAxisType|undefined;
 
+/** Cached result of the check that indicates whether the browser supports scroll behaviors. */
+let scrollBehaviorSupported: boolean|undefined;
+
 /** Check whether the browser supports scroll behaviors. */
 export function supportsScrollBehavior(): boolean {
-  return !!(typeof document == 'object' && 'scrollBehavior' in document.documentElement!.style);
+  if (scrollBehaviorSupported == null) {
+    // If we're not in the browser, it can't be supported.
+    if (typeof document !== 'object' || !document) {
+      scrollBehaviorSupported = false;
+    }
+
+    // If the element can have a `scrollBehavior` style, we can be sure that it's supported.
+    if ('scrollBehavior' in document.documentElement!.style) {
+      scrollBehaviorSupported = true;
+    } else {
+      // At this point we have 3 possibilities: `scrollTo` isn't supported at all, it's
+      // supported but it doesn't handle scroll behavior, or it has been polyfilled.
+      const scrollToFunction: Function|undefined = Element.prototype.scrollTo;
+
+      if (scrollToFunction) {
+        // We can detect if the function has been polyfilled by calling `toString` on it. Native
+        // functions are obfuscated using `[native code]`, whereas if it was overwritten we'd get
+        // the actual function source. Via https://davidwalsh.name/detect-native-function. Consider
+        // polyfilled functions as supporting scroll behavior.
+        scrollBehaviorSupported = /\{\s*\[native code\]\s*\}/.test(scrollToFunction.toString());
```

Comment:
This is a little hacky, but it seems to work. Another approach I was considering was similar to the `getRtlScrollAxisType` which creates an element and tries to scroll it which is definitely better, but I decided not to do it, because it doesn't actually detect if `scrollBehavior` is supported, but rather that the `ScrollOption` object parameter is supported. While that would fix the original issue of polyfills not being picked up, it would mean that the name of `supportsScrollBehavior` wouldn't match with what it was actually doing.